GitCommands/ExternalLinks/ExternalLinksManager.cs

```diff
@@ -2,26 +2,56 @@
 using System.Collections.Generic;
 using System.Linq;
 using GitCommands.Settings;
+using GitUIPluginInterfaces;
 
 namespace GitCommands.ExternalLinks
 {
     // NB: this implementation is stateful
     public sealed class ExternalLinksManager
     {
-        private readonly RepoDistSettings _cachedSettings;
+        private readonly ISettingsSource _cachedSettings;
         private readonly ExternalLinksManager? _lowerPriority;
         private readonly IExternalLinksStorage _externalLinksStorage = new ExternalLinksStorage();
         private readonly List<ExternalLinkDefinition> _definitions;
 
-        public ExternalLinksManager(RepoDistSettings settings)
+        public ExternalLinksManager(RepoDistSettingsSet settingsSet, SettingLevel settingLevel, bool cascade = false)
         {
-            _cachedSettings = new RepoDistSettings(null, settings.SettingsCache, settings.SettingLevel);
-            _definitions = _externalLinksStorage.Load(_cachedSettings).ToList();
+            if (settingLevel is SettingLevel.Effective)
+            {
+                cascade = true;
```

Comment:
Would it be better to throw if the user passes `settingLevel: SettingLevel.Effective, cascade: false`? It seems weird to accept an argument then ignore it.

Reply:
👍